undertow/WFLY-15452_ajp-listener_allowed_attr_pattern.adoc

```diff
@@ -0,0 +1,121 @@
+= <INSERT TITLE HERE>
+:author:            Modify ajp-listener to allow specifying AJP_ALLOWED_REQUEST_ATTRIBUTES_PATTERN
+:email:             bbaranow@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Since UNDERTOW-1667 one can set additional AJP request attribute parsing permission via env variable. However there is no way to set it in WFLY config nor directly in UndertowOptions. Later is being addressed in UNDERTOW-1978, to which this RFE is a follow up.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-15452[WFLY-15452]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/UNDERTOW-1667[UNDERTOW-1667]
+* https://issues.redhat.com/browse/UNDERTOW-1977[UNDERTOW-1977]
+* https://issues.redhat.com/browse/UNDERTOW-1978[UNDERTOW-1978]
+* https://issues.redhat.com/browse/WFLY-15453[WFLY-15453]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:msvehla@redhat.com[Martin Svehla]
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+* undertow
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [~x] OpenShift s2i
+
+* [ ] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+* Being able to configure pattern via XML configuration file.
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+== Backwards Compatibility
+
+Possibly. Subsystem transformers should be able to handle it.
+
+=== Default Configuration
```

Comment:
I think default configuration refers to what will be the default configuration of the server, can someone vouch if I'm right?
So, by default, are we having a allowed_request_attr_pattern in the server? Nope, because we don't have an ajp-listener. Still, the example of the xml could be valid in Hard Requirements, and very helpful for the test plan

Reply:
@fl4via

I prefer "standard configuration" when describing the standalone/host/domain.xml files we include in our standard distribution zip.

"Default" is ambiguous because many attributes have default values but those are only relevant is a resource with is configured. Plus, our standard configurations may have a value for an attribute that is different from its default value. So I prefer to only use 'default' to refer to default attribute behavior.

Reply:
Perhaps the default value (of the new attribute) could be undefined as in when defining the attribute definition, you don't use the `SimpleAttributeDefinitionBuilder.setDefaultValue()` function. And the meaning of that is that there is no `allowed_request_attr_pattern`, even is `ajp-listener` is defined. Or if there is a default value recommended by the specification (or a value that is commonly used by other such solutions), then we can use that as the default value.

And as for the standard configuration, if the `ajp-listener` is not defined by default, the value for `allowed_request_attr_pattern` would also be undefined regardless of what we used for `SimpleAttributeDefinitionBuilder.setDefaultValue()` 

WDYT? @fl4via @baranowb @bstansberry

Reply:
@PrarthonaPaul looking at https://github.com/wildfly/wildfly/pull/14789/files I see the attribute has no default value, so it looks like what you describe is what was implemented.

AIUI the effect of that is if the ajp-listener is configured with this feature in place the listener behavior is no different from before; i.e. undertow will reject unknown request attributes. So this feature has no 'Default configuration' compatibility concerns.

Reply:
sounds good! Thanks @bstansberry

Reply:
+1 @bstansberry @PrarthonaPaul